App crash on start

After the 13 June update, the app crash on start. After Dis-install/Re-install, it crashes when I try to choose my province.

Smartphone:

    Device: Samsung J510FN
    OS: Android 7.1.1

Thanks @GiacoMonti I'll report the stacktrace there

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: it.ministerodellasalute.immuni, PID: 7648
    java.lang.NoClassDefFoundError: java.time.OffsetDateTime
        at libcore.reflect.InternalNames.getClass(InternalNames.java:55)
        at java.lang.Class.getDexCacheType(Class.java:2551)
        at java.lang.reflect.Method.getReturnType(Method.java:150)
        at java.lang.Class.getDeclaredMethods(Class.java:1811)
        at kotlin.reflect.jvm.internal.impl.descriptors.runtime.components.ReflectClassStructure.loadMethodAnnotations(ReflectKotlinClass.kt:96)
        at kotlin.reflect.jvm.internal.impl.descriptors.runtime.components.ReflectClassStructure.visitMembers(ReflectKotlinClass.kt:90)
        at kotlin.reflect.jvm.internal.impl.descriptors.runtime.components.ReflectKotlinClass.visitMembers(ReflectKotlinClass.kt:71)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader.loadAnnotationsAndInitializers(AbstractBinaryClassAnnotationAndConstantLoader.kt:304)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader.access$loadAnnotationsAndInitializers(AbstractBinaryClassAnnotationAndConstantLoader.kt:40)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader$storage$1.invoke(AbstractBinaryClassAnnotationAndConstantLoader.kt:45)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader$storage$1.invoke(AbstractBinaryClassAnnotationAndConstantLoader.kt:44)
        at kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$MapBasedMemoizedFunction.invoke(LockBasedStorageManager.java:578)
        at kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$MapBasedMemoizedFunctionToNotNull.invoke(LockBasedStorageManager.java:651)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader.findClassAndLoadMemberAnnotations(AbstractBinaryClassAnnotationAndConstantLoader.kt:167)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader.findClassAndLoadMemberAnnotations$default(AbstractBinaryClassAnnotationAndConstantLoader.kt:146)
        at kotlin.reflect.jvm.internal.impl.load.kotlin.AbstractBinaryClassAnnotationAndConstantLoader.loadValueParameterAnnotations(AbstractBinaryClassAnnotationAndConstantLoader.kt:181)
        at kotlin.reflect.jvm.internal.impl.serialization.deserialization.MemberDeserializer$valueParameters$1$annotations$1.invoke(MemberDeserializer.kt:414)
        at kotlin.reflect.jvm.internal.impl.serialization.deserialization.MemberDeserializer$valueParameters$1$annotations$1.invoke(MemberDeserializer.kt:412)
        at kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$LockBasedLazyValue.invoke(LockBasedStorageManager.java:408)
        at kotlin.reflect.jvm.internal.impl.storage.LockBasedStorageManager$LockBasedNotNullLazyValue.invoke(LockBasedStorageManager.java:527)
        at kotlin.reflect.jvm.internal.impl.storage.StorageKt.getValue(storage.kt:42)
        at kotlin.reflect.jvm.internal.impl.serialization.deserialization.descriptors.DeserializedAnnotations.getAnnotations(DeserializedAnnotations.kt:28)
        at kotlin.reflect.jvm.internal.impl.serialization.deserialization.descriptors.DeserializedAnnotations.iterator(DeserializedAnnotations.kt:32)
        at kotlin.reflect.jvm.internal.UtilKt.computeAnnotations(util.kt:257)
        at kotlin.reflect.jvm.internal.KParameterImpl$annotations$2.invoke(KParameterImpl.kt:32)
        at kotlin.reflect.jvm.internal.KParameterImpl$annotations$2.invoke(KParameterImpl.kt:24)
        at kotlin.reflect.jvm.internal.ReflectProperties$LazySoftVal.invoke(ReflectProperties.java:92)
        at kotlin.reflect.jvm.internal.ReflectProperties$Val.getValue(ReflectProperties.java:31)
        at kotlin.reflect.jvm.internal.KParameterImpl.getAnnotations(KParameterImpl.kt)
        at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory.create(KotlinJsonAdapter.kt:233)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:97)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:67)
E/AndroidRuntime:     at com.squareup.moshi.CollectionJsonAdapter.newArrayListAdapter(CollectionJsonAdapter.java:52)
        at com.squareup.moshi.CollectionJsonAdapter$1.create(CollectionJsonAdapter.java:36)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory.create(KotlinJsonAdapter.kt:241)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at it.ministerodellasalute.immuni.logic.user.models.GreenCertificateUserJsonAdapter.<init>(GreenCertificateUserJsonAdapter.kt:20)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
        at com.squareup.moshi.internal.Util.generatedAdapter(Util.java:478)
        at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory.create(KotlinJsonAdapter.kt:179)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:97)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:67)
        at com.squareup.moshi.CollectionJsonAdapter.newArrayListAdapter(CollectionJsonAdapter.java:52)
        at com.squareup.moshi.CollectionJsonAdapter$1.create(CollectionJsonAdapter.java:36)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at it.ministerodellasalute.immuni.logic.user.models.UserJsonAdapter.<init>(UserJsonAdapter.kt:26)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
        at com.squareup.moshi.internal.Util.generatedAdapter(Util.java:478)
        at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory.create(KotlinJsonAdapter.kt:179)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:137)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:97)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:71)
        at it.ministerodellasalute.immuni.logic.user.repositories.UserRepository.save(UserRepository.kt:76)
        at it.ministerodellasalute.immuni.logic.user.UserManager.save(UserManager.kt:63)
        at it.ministerodellasalute.immuni.ui.onboarding.OnboardingViewModel.saveUserIfNeeded(OnboardingViewModel.kt:125)
        at it.ministerodellasalute.immuni.ui.onboarding.OnboardingViewModel.onProvinceSelected(OnboardingViewModel.kt:118)
        at it.ministerodellasalute.immuni.ui.onboarding.fragments.viewpager.ProvinceFragment.onClick(ProvinceFragment.kt:94)
        at it.ministerodellasalute.immuni.ui.onboarding.fragments.viewpager.ProvinceListAdapter.onItemClick(ProvinceListAdapter.kt:40)
        at it.ministerodellasalute.immuni.ui.onboarding.fragments.viewpager.ProvinceListAdapter.access$onItemClick(ProvinceListAdapter.kt:27)
        at it.ministerodellasalute.immuni.ui.onboarding.fragments.viewpager.ProvinceListAdapter$ProvinceVH$1.invoke(ProvinceListAdapter.kt:49)
        at it.ministerodellasalute.immuni.ui.onboarding.fragments.viewpager.ProvinceListAdapter$ProvinceVH$1.invoke(ProvinceListAdapter.kt:44)
        at it.ministerodellasalute.immuni.extensions.view.SafeClickListenerKt$setSafeOnClickListener$safeClickListener$1.invoke(SafeClickListener.kt:40)
        at it.ministerodellasalute.immuni.extensions.view.SafeClickListenerKt$setSafeOnClickListener$safeClickListener$1.invoke(SafeClickListener.kt)
        at it.ministerodellasalute.immuni.extensions.view.SafeClickListener.onClick(SafeClickListener.kt:34)
        at android.view.View.performClick(View.java:5637)
        at android.view.View$PerformClick.run(View.java:22429)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "java.time.OffsetDateTime" on path: DexPathList[[zip file "/data/app/it.ministerodellasalute.immuni-1/base.apk"],nativeLibraryDirectories=[/data/app/it.ministerodellasalute.immuni-1/lib/x86, /data/app/it.ministerodellasalute.immuni-1/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at libcore.reflect.InternalNames.getClass(InternalNames.java:53)
        	... 77 more
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -61,6 +61,7 @@ android {
     }
 
     compileOptions {
+        coreLibraryDesugaringEnabled true
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
@@ -110,6 +111,7 @@ dependencies {
     implementation project(':network')
     implementation project(':decoder1.0.0')
     implementation fileTree(dir: 'libs', include: ['*.jar', '*.aar'])
+    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
 
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesCoreVersion"

